modules/andes-core/broker/src/main/java/org/wso2/andes/server/ack/LimitedSizeQueueEntryHolder.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.andes.server.ack;
+
+import org.apache.log4j.Logger;
+import org.wso2.andes.AMQException;
+import org.wso2.andes.amqp.AMQPUtils;
+import org.wso2.andes.amqp.QpidAndesBridge;
+import org.wso2.andes.configuration.AndesConfigurationManager;
+import org.wso2.andes.configuration.enums.AndesConfiguration;
+import org.wso2.andes.server.AMQChannel;
+import org.wso2.andes.server.message.AMQMessage;
+import org.wso2.andes.server.queue.QueueEntry;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class LimitedSizeQueueEntryHolder extends LinkedHashMap<Long, QueueEntry> {
+
+    protected static final Logger _logger = Logger.getLogger(LimitedSizeQueueEntryHolder.class);
+
+    private final int limit = AndesConfigurationManager.readValue
```

Comment:
and may be move it to the constructor. so the initialization is uniform across this class.